net/dns/wsl_windows.go

```diff
@@ -66,17 +45,17 @@ func wslDistros() ([]string, error) {
 	return distros, nil
 }
 
-func decodeUTF16(b []byte) (string, error) {
+func decodeUTF16(b []byte, endian binary.ByteOrder) ([]byte, error) {
 	if len(b) == 0 {
-		return "", nil
+		return nil, nil
 	} else if len(b)%2 != 0 {
-		return "", fmt.Errorf("decodeUTF16: invalid length %d", len(b))
+		return nil, fmt.Errorf("decodeUTF16: invalid length %d", len(b))
 	}
 	var u16 []uint16
 	for i := 0; i < len(b); i += 2 {
-		u16 = append(u16, uint16(b[i])+(uint16(b[i+1])<<8))
+		u16 = append(u16, endian.Uint16(b[i:]))
 	}
-	return string(utf16.Decode(u16)), nil
+	return []byte(string(utf16.Decode(u16))), nil
```

Comment:
good thing this isn't a hot path :)

Reply:
Yeah, the unicode encoding libraries don't make this transcode fun to do :(